modules/codegen/src/main/scala/com/twilio/guardrail/extract/package.scala

```diff
@@ -34,6 +34,9 @@ package object extract {
   def PackageName[F: VendorExtension.VendorExtensible](v: F, vendorPrefixes: List[String]): Option[String] =
     extractFromNames[F, String](v, vendorPrefixes.map(_ + "-package"))
 
+  def ClassPrefix[F: VendorExtension.VendorExtensible](v: F, vendorPrefixes: List[String]): Option[String] =
+    extractFromNames[F, String](v, vendorPrefixes.map(_ + "-class-prefix"))
```

Comment:
how do we move from here? :-) @kelnos

Reply:
Ooof, so sorry @zezutom, I somehow completely missed this.

I think it can't hurt to allow it, though the use-case isn't all that clear to me.  We're always going to apply the language-specific name formatting rules, so if it's normal somewhere to have `some_foo_client` as a type name, and someone does `x-class-prefix: SomeFoo`, we'll transform the name.

I think we should actually do something slightly more flexible, and even add this flexibility to all our existing vendor extension extractors: we should accept a vendor extension without the vendor prefix.  So for this one, Scala might look first for `x-scala-class-prefix`, then `x-jvm-class-prefix`, and then finally fall back to `x-class-prefix`.  Can't hurt to do this for the other ones too; e.g. `x-scala-type` -> `x-jvm-type` -> `x-type`.

Reply:
(But given that this PR has been languishing [my fault], I'm happy merging it as-is.)

Reply:
Hi @kelnos , no problem at all and thanks for all the feedback. If you merge this one, I'll attempt to apply your suggestions in a new PR ;-)